ecl/hql/hqlgram2.cpp

```diff
@@ -12311,14 +12311,21 @@ void parseAttribute(IHqlScope * scope, IFileContents * contents, HqlLookupContex
         throw;
     }
     OwnedHqlExpr parsed = scope->lookupSymbol(name, LSFsharedOK|LSFnoreport, ctx);
-    bool canCache = parsed && (parsed->getOperator() != no_forwardscope) && !alreadySimplified;
-    bool isMacro = parsed && parsed->isMacro();
+    bool canCache = false;
+    bool isMacro = false;
+    bool neverSimplify = false;
+    if (parsed)
+    {
+        bool canCache = (parsed->getOperator() != no_forwardscope) && !alreadySimplified ;
```

Comment:
This is declaring a new variable - so the outer one will never be modified.

Reply:
oops